src/main/java/adris/altoclef/commands/GotoCommand.java

```diff
@@ -13,14 +13,19 @@
     private static final int EMPTY = -1;
 
     public GotoCommand() throws CommandException {
-        super("goto", "Tell bot to travel to a set of coordinates.", new Arg(Integer.class, "x"), new Arg(Integer.class, "y", EMPTY, 2), new Arg(Integer.class, "z"));
+        super("goto", "Tell bot to travel to a set of coordinates.", new Arg(String.class, "dimension", String.class, 0, false), new Arg(Integer.class, "x"), new Arg(Integer.class, "y", EMPTY, 2), new Arg(Integer.class, "z"));
```

Comment:
Does anyone know how i can make the last argument (z) optional. I can't figure out this handler

Reply:
Why would you want to make the z optional? Are you trying to make GotoCommand work with only a y value?

The system I have implemented is supremely jank, but it could be `new Arg(Integer.class, "z", EMPTY, 1)` or something like that? Will need to overhaul the command system as well to make this work better, switching over to something similar to what Baritone uses.

Reply:
(I believe the higher the number at the end, the earlier that parameter will use its default value. So the dimension parameter should probably be given 2 instead of 0, as you'd want to assume that by default.)

So sorry about this crappy system lol.

Also you can use Enums here, I'd suggest using the "Dimension" enum and setting the default value to null. When parsing the commands, if the dimension is null it can be ignored.

Reply:
This is really confusing... I've basically never used Java before but I'll try fixing this when I get home